library/src/main/java/com/android/volley/RequestQueue.java

```diff
@@ -343,6 +356,8 @@ void finish(Request<?> request) {
                 }
             }
         }
+
+        mRequestLogger.log(request);
```

Comment:
The problem is that you're controlling the `Request` from outside. `RequestQueue` knows too much about it. It breaks the [tell, don't ask](http://martinfowler.com/bliki/TellDontAsk.html) principle, but I guess that's too late to worry about that now